docs/products/postgresql/concepts/pg-shared-buffers.rst

```diff
@@ -0,0 +1,166 @@
+
+  CREATE EXTENSION pg_buffercache;
+  CREATE EXTENSION
+
+The following query will calculate how many blocks from tables (r), indexes (i), sequences (S), and other objects are currently cached:
+
+.. code-block:: shell
+
+  SELECT c.relname, c.relkind
+    , pg_size_pretty(count(*) * 8193) as buffered
+    , round(100.0 * count(*) / ( SELECT setting FROM pg_settings WHERE name='shared_buffers')::integer,1) AS buffers_percent
+    , round(100.0 * count(*) * 8192 / pg_relation_size(c.oid),1) AS percent_of_relation
+  FROM pg_class c
+  INNER JOIN pg_buffercache b ON b.relfilenode = c.relfilenode
+  INNER JOIN pg_database d ON (b.reldatabase = d.oid AND d.datname = current_database())
+  WHERE c.oid >= 16384
+  AND pg_relation_size(c.oid) > 0
+  GROUP BY c.oid, c.relname
+  ORDER BY 3 DESC
+  LIMIT 10;
+
+   relname | relkind | buffered | buffers_percent | percent_of_relation
+  ---------+---------+----------+-----------------+---------------------
+   records | r       | 781 MB   |            99.7 |                27.2
+
+Relations with object ID below `16384` are reserved system objects.
+
+Inspecting query cache performance
+----------------------------------
+Queries can also be inspected for cache hit performance using `EXPLAIN`:
+
+.. code-block:: shell
+  
+  EXPLAIN (ANALYZE, BUFFERS, VERBOSE)
+  defaultdb-> SELECT * from records;
+                                                            QUERY PLAN
+  --------------------------------------------------------------------------------------------------------------------------------
+  Seq Scan on public.records  (cost=0.00..480095.20 rows=11207220 width=77) (actual time=0.158..16863.051 rows=11600000 loops=1)
+    Output: id, "timestamp", data
+    Buffers: shared hit=92345 read=275678 dirtied=10938
+  Query Identifier: 2582883386000135492
+  Planning:
+    Buffers: shared hit=30 dirtied=2
+  Planning Time: 1.081 ms
+  Execution Time: 17467.342 ms
+  (8 rows)
+
+Using `hit / (hit + read)` shows `~25%` of this full table scan was in the ``shared_buffers``
+
+Putting data into cache manually
+--------------------------------
+Occasionally, it may be desirable to prewarm the ``shared_buffers`` in anticipation of a specific workload, such as a large analytical query set used for reporting.
+This can be accomplished using the ``pg_prewarm`` extension.
+
+.. code-block:: shell
+
+  CREATE EXTENSION pg_prewarm;
+  CREATE EXTENSION
+
+In the simplest operation, call the ``pg_prewarm`` function and pass the name of the desired table.
+
+.. code-block:: shell
+
+ SELECT * FROM pg_prewarm('public.records');
+ pg_prewarm
+ ------------
+      368023
+
+  SELECT pg_size_pretty(pg_relation_size('public.records'));
+  pg_size_pretty
+  ----------------
+  2875 MB
+
+In this example, 368023 pages have be read into the cache (or ~2875 MB).
+
+If the ``shared buffers`` size is less than pre-loaded data, only the tailing end of the data will be cached as the earlier data will encounter forced ejection.
```

Comment:
the tailing end of the data will be => the tailing end of the data is
the earlier data will encounter forced ejection => the earlier data encounters a forced ejection.